contracts/Aavegotchi/ForgeDiamond/facets/ForgeVRFFacet.sol

```diff
@@ -0,0 +1,249 @@
+    event GeodeEmpty(address user, uint256 geodeTokenId, bytes32 requestId, uint256 blockNumber);
+    event GeodeRefunded(address user, uint256 geodeTokenId, bytes32 requestId, uint256 blockNumber);
+
+    function forgeTokenFacet() internal view returns (ForgeTokenFacet facet) {
+        facet = ForgeTokenFacet(address(this));
+    }
+
+    function forgeFacet() internal view returns (ForgeFacet facet) {
+        facet = ForgeFacet(address(this));
+    }
+
+    // Maximum number of geodes that can be opened in one call.
+    uint256 constant MAX_VRF = 15;
+
+    function linkBalance() external view returns (uint256 linkBalance_) {
+        linkBalance_ = s.link.balanceOf(address(this));
+    }
+
+    function vrfCoordinator() external view returns (address) {
+        return s.vrfCoordinator;
+    }
+
+    function link() external view returns (address) {
+        return address(s.link);
+    }
+
+    function keyHash() external view returns (bytes32) {
+        return s.keyHash;
+    }
+
+    function getMaxVrf() external view returns (uint256) {
+        return MAX_VRF;
+    }
+
+    function areGeodePrizesAvailable() public view returns (bool) {
+        for (uint256 i; i < s.geodePrizeTokenIds.length; i++) {
+            uint256 tokenId = s.geodePrizeTokenIds[i];
+
+            if (tokenId != 0 && s.geodePrizeQuantities[tokenId] > 0 && s.ownerItemBalances[address(this)][tokenId] > 0) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function numTotalPrizesLeft() public view returns (uint256 total) {
+        for (uint256 i; i < s.geodePrizeTokenIds.length; i++) {
+            total += s.geodePrizeQuantities[s.geodePrizeTokenIds[i]];
+        }
+    }
+
+    function openGeodes(uint256[] calldata _geodeTokenIds, uint256[] calldata _amountPerToken) external whenNotPaused {
+        require(areGeodePrizesAvailable(), "ForgeVRFFacet: No prizes currently available");
+        require(_geodeTokenIds.length == _amountPerToken.length, "ForgeVRFFacet: mismatched arrays");
+
+        uint256 totalGeodes;
+        for (uint256 i; i < _amountPerToken.length; i++) {
+            totalGeodes += _amountPerToken[i];
+        }
+        require(totalGeodes <= MAX_VRF, "ForgeVRFFacet: Exceeds max total geodes per call");
+
+        address sender = LibMeta.msgSender();
+        uint256 total;
+        for (uint256 i; i < _amountPerToken.length; i++) {
+            total += _amountPerToken[i];
+
+            require(_geodeTokenIds[i] >= GEODE_COMMON && _geodeTokenIds[i] <= GEODE_GODLIKE, "ForgeVRFFacet: Invalid geode token ID");
+            require(forgeTokenFacet().balanceOf(sender, _geodeTokenIds[i]) >= _amountPerToken[i], "ForgeVRFFacet: not enough geodes owned");
+        }
+
+        // spend geodes
+        for (uint256 i; i < _geodeTokenIds.length; i++) {
+            forgeTokenFacet().safeTransferFrom(sender, address(this), _geodeTokenIds[i], _amountPerToken[i], "");
```

Comment:
The geodes don't get burned? Just sent to the forge diamond? @infiniRekt

Reply:
Ah I see, because they are refunded if there are no prizes left

Reply:
In that case maybe you would want to burn the geode in the claimWinnings function?